### ISSUE TITLE
Problems with Select

### ISSUE THREAD

**Issue by vgirard** [stirno/FluentAutomation#65]
I have a problem when I try to select a value in a dropdown by default you say it's the Value but it use the Text. If I do this : I.Select(Option.Value, type).From("#Type"); it works but this does not : I.Select(type).From("#Type");

The other issu I have is when I check the selected value I get all the possible values and not only the selected one : I.Expect.Value("Qc").In("#Province"); returns FluentAutomation.Exceptions.FluentExpectFailedException : Expected SelectElement [#Province] selected option value to be [Qc] but it was actually [Alberta
Colombie-Britannique
Manitoba
Nouveau-Brunswick
Terre-Neuve
Nouvelle-Écosse
Territoire du Nord Ouest
Nunavut
Ontario
Île du Prince Édouard
Québec
Saskatchewan
Yukon].

I'm on the latest version from Nuget and I like it so far :)

**Comment by stirno** [comment]
Just saw this. I'll look into this tomorrow morning and get back to you ASAP. We used to have some behaviors to fallback to Text based matches when values weren't set for options in selects. Thats been removed in v2 though so I wouldn't expect that now.

Do you have some sample HTML I can repro with? Helpful but not necessary. Thanks!

**Comment by vgirard** [comment]
What do you need as sample HTML? Does this help : 

```
 <select id="Province" name="Province"><option></option>
 <option value="AB">Alberta</option>
 <option value="BC">Colombie-Britannique</option> 
 <option value="MB">Manitoba</option>
 <option value="NB">Nouveau-Brunswick</option>
 <option value="NL">Terre-Neuve</option>
 <option value="NS">Nouvelle-&#201;cosse</option>
 <option value="NT">Territoire du Nord Ouest</option>
 <option value="NU">Nunavut</option>
 <option value="ON">Ontario</option>
 <option value="PE">&#206;le du Prince &#201;douard</option>
 <option value="QC">Qu&#233;bec</option>
 <option value="SK">Saskatchewan</option>
 <option value="YT">Yukon</option>
 </select>
```

**Comment by stirno** [comment]
Yes indeed. Saves me some time. Thanks!

**Comment by stirno** [comment]
Ok so 2 bugs confirmed.
1. with the intellisense/comments for the I.Select() commands. I'll fix it to match the intended behavior.
2. as you noticed, theres an issue with processing expects against selects. I'll get this into source fixed tonight and tag it for 2.0.1 release to NuGet.

Thanks for the report

**Comment by stirno** [comment]
Decided to push this out to NuGet. Its up as v2.0.0.2 now. Please update and let me know if you have any further questions. Thanks!

**Comment by vgirard** [comment]
Thanks for the quick fix, it works great!

But the Expect on the Text seems to have the same problem that the onde on the Value. If I do I.Expect.Text("Québec").In("#Province"); it returns the list of all the text options.

FluentAutomation.Exceptions.FluentExpectFailedException : Expected SelectElement [#Province] selected option text to be [Québec] but it was actually [Alberta
Colombie-Britannique
Manitoba
Nouveau-Brunswick
Terre-Neuve
Nouvelle-Écosse
Territoire du Nord Ouest
Nunavut
Ontario
Île du Prince Édouard
Québec
Saskatchewan
Yukon].

**Comment by stirno** [comment]
True story. I'm kinda surprised I didn't update that bit as well. Scope-lock on the reported problem I guess. I'll fix that in a bit. Re-opening until I make it right.

**Comment by edutoporoski** [comment]
I have the same problem, I found that when the value of value = "" is equal to the value between the tag <option> (I.Expect.Value ("Quebec").In("Provice");) Works, but, when the value between value="" differs the tag option does not work.

ex:. <option> value="QC"> Québec <\ option> does not work.
         <option> value="Québec"> Québec <\ option> works

**Comment by stirno** [comment]
@edutoporoski Thats working as intended. The I.Expect.Value functionality will always look at the value tag. I.Expect.Text looks at the actual text tag.

So in your scenario, the proper expect would be:

```
<select><option value="QC">Québec</option></select>

I.Expect.Text("Québec").In("select");
```

or

```
<select><option value="QC">Québec</option></select>

I.Expect.Value("QC").In("select");
```

Long ago in v1 we tried to do 'dynamic' fallbacks where if we failed to find a match on Text, we'd check Value. It introduced some issues with certain apps so we decided to make it strict/explicit. It checks what its asked to check.

**Comment by pauljz** [comment]
To @vgirard's original issue, in 2.x `I.Assert.Text("Québec").In("#Province");` was performing the assertion against the text of all the values. This will be fixed in 3.0.0 to only assert against the text of the selected option(s).

**Comment by stirno** [comment]
More accurately.... it was just being stupid and using the default .Text property. My bad. And by will be fixed, he means its checked in for 3.0

**Comment by lovisharma** [comment]
Hi, 

I just want to confirm that -  
1. This is suitable for Firefox, Chrome, All IE versions and safari without any issue when finding element or something other as i am facing issues in IE with selenium C# ?  
2. Can we find all elements on the page using CSS selector, as it only supports CSS selector? Is there any scope of finding element by Xpath or by NAME, ID as in selenium ?
3. Is there any slowness issue in IE that same we are facing with selenium c#?

Please reply for above queries, so that we can go for this ?
